packages/flow-core/src/components/f-file-upload/f-file-upload.ts

```diff
@@ -0,0 +1,536 @@
+
+	/**
+	 * @attribute assigned elements inside slot description
+	 */
+	@queryAssignedElements({ slot: "description" })
+	_descriptionNodes!: NodeListOf<HTMLElement>;
+
+	/**
+	 * @attribute flag to check if slot description is present or not
+	 */
+	@state()
+	_hasDescription = false;
+
+	/**
+	 * @attribute assigned elements inside slot help
+	 */
+	@queryAssignedElements({ slot: "help" })
+	_helpNodes!: NodeListOf<HTMLElement>;
+
+	/**
+	 * @attribute flag to check if slot help is present or not
+	 */
+	@state()
+	_hasHelperText = false;
+
+	selectedFiles?: File[] = [];
+
+	fileInputRef: Ref<HTMLInputElement> = createRef();
+
+	/**
+	 * on click open file selector window on OS
+	 */
+	handleClick() {
+		if (!this.loading) {
+			const fileInput = this.fileInputRef.value as HTMLInputElement;
+			fileInput.value = "";
+			fileInput.click();
+			this.acceptedFilesFlag = true;
+			this.sizeLimitFlag = true;
+		}
+	}
+
+	/**
+	 * @param e DragEvent
+	 * drag and drop file(s) and event emission
+	 */
+	dropFile(e: DragEvent) {
+		if (!this.loading) {
+			e.preventDefault();
+			const files = e.dataTransfer?.files;
+			const filesArr = files ? Array.from(files) : [];
+			if (this.fileType !== "all") {
+				this.acceptedFilesFlag = filesArr.every(
+					item => item.type && this.fileType?.includes(item.type.split("/")[1])
+				);
+			}
+			this.sizeLimitFlag =
+				this.maxSize && this.bytes ? filesArr.every(item => item.size < this.bytes) : true;
+			if (this.acceptedFilesFlag && this.sizeLimitFlag) {
+				if (this.type === "multiple") {
+					filesArr.forEach(item => {
+						this.selectedFiles?.push(item);
+					});
+				} else {
+					if (filesArr?.length === 1) {
+						this.selectedFiles = filesArr;
+					}
+				}
+				this.dispatchOnInput(e);
+			}
+		}
+	}
+
+	/**
+	 * select file(s) from browse options
+	 * @param e InputEvent
+	 */
+	selectFile(e: InputEvent) {
+		const fileInput = this.fileInputRef.value as HTMLInputElement;
+		const files = fileInput.files;
+		const filesArr = files ? Array.from(files) : [];
+		this.sizeLimitFlag =
+			this.maxSize && this.bytes ? filesArr.every(item => item.size < this.bytes) : true;
+		if (this.sizeLimitFlag) {
+			if (this.type === "multiple") {
+				filesArr.forEach(item => {
+					this.selectedFiles?.push(item);
+				});
+			} else {
+				this.selectedFiles = filesArr;
+			}
+			this.dispatchOnInput(e);
+		}
+	}
+
+	/**
+	 * dispatch input event
+	 * @param e Event
+	 */
+	dispatchOnInput(e: Event) {
+		e.stopPropagation();
+		/**
+		 * @event input
+		 */
+		const event = new CustomEvent("input", {
+			detail: {
+				value:
+					this.type === "single"
+						? this.selectedFiles
+							? this.selectedFiles[0]
+							: []
+						: this.selectedFiles
+			},
+			bubbles: true,
+			composed: true
+		});
+		this.value =
+			this.type === "single"
+				? this.selectedFiles
+					? this.selectedFiles[0]
+					: []
+				: this.selectedFiles;
+		this.dispatchEvent(event);
+		this.requestUpdate();
```

Comment:
you don't request update here, since you are assigning to this.value which itself is reactive